### PR TITLE
new form pages

### LINKED ISSUE #88: [FE] Create Form page
### Description
Users want to be able to create forms for dogs.

**Add Form button**
- on the Forms tab there is a **Add Form** dropdown button allowing you to select a form type to create
- create this component
- onClick have the button link to `/dogs/[id]/forms/new?type=FormType` where the query parameter is based on the form type selected
    - notice that there are 3 form types (MonthlyPlaced, MonthlyUnplaced, VolunteerInteraction), but only two options in the dropdown. This is because the monthly form types are implicitly chosen based on if the dog has `location: "Placed"`. Make sure this functionality is incorporated into the button.

**Create form page**
- create the page `pages/dogs/[id]/forms/new.jsx`
- use the query param `type` to determine what form template to render
- I suggest using `react-hook-form`
- on **Save**, perform client-side validation
- if valid, make a request to `POST /api/forms`
- on **Cancel**, discard all form state and navigate back to individual dog page
- create toasts for certain events (see Figma design)

### Acceptance Criteria
- [x] Proper error handling
- [x] Implements Figma designs
- [x] Proper JS doc comments above functions

### Other Notes
- N/a

### PR DESCRIPTION
**Pull request by siri-star**
## New Form Pages
Issue Number(s): #88, #87

What does this PR change and why?
This change allows users to add a new form (of which there are 3 types!).

### Checklist
- [X] Requirements have been implemented
- [X] Acceptance criteria is met
- [X] Database schema [docs](https://www.notion.so/gtbitsofgood/Database-Schema-Docs-a841a38d02db4428a7cdc85dcbb8a35c?pvs=4) have been updated or are not necessary
- [X] Code follows design and style guidelines
- [X] Commits follow guidelines (concise, squashed, etc)
- [X] Relevant reviewers (EM) have been assigned to this PR

### Related PRs
PR Number(s): #92 

### How To Test

1. Navigate to `/dogs/[id]/forms/new?type=volunteerInteraction` (or monthlyPlaced / monthly Unplaced).
2. Ensure that the correct questions and response types are rendered.
3. Enter answers.
4. When submitting the form, ensure that (1) the form object appears in the database or (2) the appropriate error messages surface (as described by the Figma).

### CODE REVIEW

File: src/pages/dogs/[id]/forms/new.jsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import ConfirmCancelModal from "@/components/ConfirmCancelModal";
+import { ChevronLeftIcon } from "@heroicons/react/24/solid";
+import Link from "next/link";
+import {
+  MONTHLY_PLACED_FORM,
+  MONTHLY_UNPLACED_FORM,
+  VOLUNTEER_FORM,
+} from "@/utils/formConsts";
+
+export default function NewForm() {
```
Comment (afazio1) [inline src/pages/dogs/[id]/forms/new.jsx:4]: Make this a general Form component. You can have the component take in an action enum (view, edit, new) and based on that alter the inputs (disabled or not) + change the submit button logic

File: src/pages/dogs/[id]/forms/new.jsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  if (!dog || dog === undefined || !dog.success) {
+    return <div>loading</div>;
+  }
+
+  const name = dog.data.name;
+
+  return (
+    <div>
+      {showCancelModal
+        ? ConfirmCancelModal(
```
Comment (afazio1) [inline src/pages/dogs/[id]/forms/new.jsx]: Great the you made this reusable across many pages

**Review by afazio1 — CHANGES_REQUESTED** [review]

File: src/pages/dogs/[id]/forms/new.jsx (hunk 1 above)
Comment (siri-star) [reply src/pages/dogs/[id]/forms/new.jsx:4]: done!
I refactored the newForm stuff into a general form component and added the view / add logic. [the edit logic still needs to be implemented]

File: src/components/Form.jsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+  const { data: session } = useSession();
+
+  let form = {};
+  form.user = session?.user._id;
+  form.dog = router.query.id;
+
+  let formType = VOLUNTEER_FORM;
```
Comment (afazio1) [inline src/components/Form.jsx]: use the `formMap` in utils for this

File: src/pages/dogs/[id]/forms/new.jsx [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import Form from "@/components/Form";
+
+export default function NewForm() {
+  return Form("view");
```
Comment (afazio1) [inline src/pages/dogs/[id]/forms/new.jsx]: i think make this "new" because "view" action is different (we have to populate the form with dog info)

**Review by afazio1 — APPROVED** [review]

**Pull request merged**